### PR TITLE
Assignment 2

### PR DESCRIPTION
**Pull request by alexfcab**

### CODE REVIEW

File: Assignment2/alexfcab-fromJSONtoJSON-LD.jsonld [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{
+    "@context": {
+        "person": "https://schema.org/Person",
+        "rdfs": "http://www.w3.org/2000/01/rdf-schema#",
+        "fullName": "rdfs:label",
+        "hometown": "rdfs:label",
```
Comment (ocorcho) [inline Assignment2/alexfcab-fromJSONtoJSON-LD.jsonld]: These cannot be rdfs:label. The same with other properties

**Review by ocorcho — CHANGES_REQUESTED** [review]

**Comment by ocorcho** [comment]
JSON-LD is not fixed. The rest ok
Now to @dgarijo for assignment 4.

**Comment by alexfcab** [comment]
Is JSON-LD fixed now?

File: Assignment4/alexfcab-200385/task06.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1h0fyE_7g4Y9jYROMCdizes5Qa-6RbvHK
+
+**Task 06: Modifying RDF(s)**
+"""
+
+!pip install rdflib
```
Comment (dgarijo) [inline Assignment4/alexfcab-200385/task06.py]: This won't work in Python. It's notebook magic

File: Assignment4/alexfcab-200385/task06.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+g.add((ns.Jane, vcard.Given, Literal("Jane")))
+g.add((ns.Smith, vcard.Family, Literal("Smith")))
+for s,p,o in g:
+  print(s,p,o)
+
+"""**TASK 6.5: Add UPM as the university where John Smith works**"""
+
+# Visualize the results
+g.add((ns.UPM, RDF.type, ns.University))
+g.add((ns.JohnSmith, vcard.work, ns.UPM))
```
Comment (dgarijo) [inline Assignment4/alexfcab-200385/task06.py]: This property does not exist in vcard

File: Assignment4/alexfcab-200385/task07.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "Person" with RDFLib and SPARQL**"""
+
+#RDFLib
+ns = Namespace("http://somewhere#")
+for s, p, o in g.triples((None, RDFS.subClassOf, ns.Person)):
+  print(s, "is subclass of ", o)
+  for s2, p2, o2 in g.triples((None, RDFS.subClassOf, s)):
+    print(s2, "is subclass of", s, "and subclass of", o)
```
Comment (dgarijo) [inline Assignment4/alexfcab-200385/task07.py]: second loop is not needed

File: Assignment4/alexfcab-200385/task07.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+print("============================")
+#SPARQL
+from rdflib.plugins.sparql import prepareQuery
+q1 = prepareQuery('''
+  SELECT DISTINCT ?x
+  WHERE{
+	  ?x (rdfs:subClassOf/rdfs:subClassOf*) ns:Person
```
Comment (dgarijo) [inline Assignment4/alexfcab-200385/task07.py]: only 1 subclassOf is ok

**Review by dgarijo — CHANGES_REQUESTED** [review]
Small changes requested

**Comment by alexfcab** [comment]
@dgarijo could you please review it again, I made some changes according to your annotations

File: Assignment4/alexfcab-200385/task06.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+"""**TASK 6.5: Add UPM as the university where John Smith works**"""
+
+g.add((ns.UPM, RDF.type, ns.University))
+
+g.add((ns.worksAt, RDFS.domain, ns.Person))
+g.add((ns.worksAt, RDF.type, ns.Property))
+g.add((ns.worksAt, RDFS.range, ns.University))
+
+g.add((ns.JohnSmith, vcard.worksAt, ns.UPM))
```
Comment (dgarijo) [inline Assignment4/alexfcab-200385/task06.py]: worksAt does not exist neither in vcard. Create your own under the ns  namespace

**Review by dgarijo — CHANGES_REQUESTED** [review]
mini changes needed

**Comment by alexfcab** [comment]
@dgarijo is task06 okay now? same with jsontojsonld @ocorcho

**Review by dgarijo — APPROVED** [review]

**Comment by dgarijo** [comment]
@ocorcho assignment 4 is now ok. Please merge when appropriate

**Pull request merged**